### ISSUE TITLE
please add mdx support

### ISSUE THREAD

**Issue by diomed** [satnaing/astro-paper#131]
MDX allows you to [use variables, JSX expressions and components within Markdown content](https://docs.astro.build/en/guides/markdown-content/#mdx-only-features) in Astro.

More on it:
https://github.com/withastro/astro/tree/main/packages/integrations/mdx#why-mdx

**Comment by fchrapek** [comment]
Hi @diomed, Astro has really decent documentation. Just follow the steps here: https://docs.astro.build/en/guides/integrations-guide/mdx/ and you should be good to go.

**Comment by diomed** [comment]
unsure why it wouldnt come out of the box

**Comment by fchrapek** [comment]
@diomed Hmm, perhaps MDX might be an overkill if you want a simple blog with just text + images. Especially when Astro supports MD images optimization. Maybe that was the initial idea.

Let's wait for the @satnaing to respond.

To be clear, I'm with you when it comes to using MDX. Is some contexts, MD is too limited.

**Comment by diomed** [comment]
@fchrapek  hey , are you on Discord?

**Comment by fchrapek** [comment]
@diomed nope

**Comment by satnaing** [comment]
Just like @fchrapek mentioned, I think it is a little bit overkill for average users who simply want to display content in their blog posts.
Initially, AstroPaper was intended to be a minimal blog template without extensive features, which is why I didn't include MDX support by default.
Moreover, I assume the template users can easily integrate MDX themselves if they wish.

However, there are two options to address this issue:

1) add a blog post (docs) for integrating AstroPaper with MDX
2) add MDX out of the box

What do you think?

**Comment by diomed** [comment]
you could replace old blog post for forestry cms with this one, since forestry doesnt exist anymore. unless it's just updatable for tina cms ¯\_(ツ)_/¯

**Comment by paultman** [comment]
Agreed that it's nice to have astro-paper lean, yet extendable via integrations already built for astro.
Personally I prefer markdoc over MDX, however as we all have different tools, I'm happier setting it up myself, or using an .astro page over .md when needed.

**Comment by mdmrk** [comment]
I tried adding MDX following astro docs, but it only renders components in some builds processes. it looks like some kind of race condition to me.
